### PR TITLE
[T2 FLOSS] Improvements to the api doc

### PR DESCRIPTION
**Pull request by muskater**
## [FLOSS T2]
With this PR we would like to improve core API documentation.
_Who did which parts of this task:_ 
We discussed areas where the documentation could be improved together as team. Then each team member picked on of the topics and created a PR.
_On which CI systems the tests run_
The same as in T1.

---
## Basics
This is still work in progress: We want to improve some wording and layout of the API

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Review by flo91 — COMMENTED** [review]
**[FLOSS T2]**

Thanks for the PR!

It contains some small but useful improvements for the C-API documentation (as code comments). No actual code was changed in this PR, so that shouldn't introduce any problems.

A nice improvement would be to combine ("squash") the three commits into one,
apart from that the commit messages look good.

Currently, the build server reports an error because of a manpage to change, but if the PR is rebased with current master, that problem should probably be solved.

File: src/libs/loader/doc.c [hunk 1]
```diff
@@ -40,6 +40,7 @@
  * @return -1 on error
  * @return >=0 otherwise
  * @ingroup modules
+ * @post the key system:/elektra/modules is added in case of success
```
Comment (markus2330) [inline src/libs/loader/doc.c:43]: good catch!

**Review by markus2330 — APPROVED** [review]
very good changes

**Comment by markus2330** [comment]
> A nice improvement would be to combine ("squash") the three commits into one, apart from that the commit messages look good.

It is also perfectly fine to keep the commits separated.

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Please rebase and set ready to merge once it is ready.

**Comment by muskater** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Thank you! :fireworks:

**Pull request merged**